### PR TITLE
Bicaws 2266 separate triggers and exceptions locking

### PR DESCRIPTION
**Pull request by emadkaramad**
This PR updates the locking logic to lock exceptions and triggers separately based on the user groups.

### CODE REVIEW

File: src/features/CourtCaseDetails/CourtCaseDetails.stories.tsx [hunk 1]
```diff
@@ -63,3 +63,22 @@ DetailsLockedByAnotherUser.play = ({ canvasElement }) => {
...
+
+export const TriggersNotVisibleToUser: ComponentStory<typeof CourtCaseDetails> = () => (
+  <CourtCaseDetails courtCase={courtCase} lockedByAnotherUser={false} aho={aho} triggersVisible={false} />
+)
+
+TriggersNotVisibleToUser.play = async ({ canvasElement }) => {
+  const canvas = within(canvasElement)
+  await expect(canvas.getByText(courtCase.ptiurn)).toBeInTheDocument()
+  await expect(canvas.getByText(courtCase.courtName)).toBeInTheDocument()
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (jaskaransarkaria) [inline src/features/CourtCaseDetails/CourtCaseDetails.stories.tsx:75]: iirc I turned this eslint error off so I don't think we need this anymore

File: src/services/getCourtCase.ts [hunk 2]
```diff
@@ -2,10 +2,7 @@ import { DataSource, EntityManager } from "typeorm"
 import CourtCase from "./entities/CourtCase"
 import PromiseResult from "../types/PromiseResult"
 
-const getCourtCase = async (
-  dataSource: DataSource | EntityManager,
-  courtCaseId: number
-): PromiseResult<CourtCase | null> => {
+const getCourtCase = (dataSource: DataSource | EntityManager, courtCaseId: number): PromiseResult<CourtCase | null> => {
```
Comment (jaskaransarkaria) [inline src/services/getCourtCase.ts:5]: How come the async is removed here 🤔

File: src/services/tryToLockCourtCase.ts [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
 
 const tryToLockCourtCase = (
   dataSource: DataSource,
   courtCaseId: number,
-  userName: string,
-  lockException = true,
-  lockTrigger = true
+  user: User
 ): Promise<UpdateResult | Error> | Error => {
+  const lockException = user.canLockExceptions
```
Comment (jaskaransarkaria) [inline src/services/tryToLockCourtCase.ts:11]: Can we just destructure this from user on line 9 so we don't need these variables?

File: src/services/unlockCourtCase.ts [hunk 4]
```diff
@@ -1,19 +1,29 @@
 import { DataSource, UpdateResult } from "typeorm/"
+import { QueryDeepPartialEntity } from "typeorm/query-builder/QueryPartialEntity"
 import CourtCase from "./entities/CourtCase"
+import User from "./entities/User"
 
-const unlockCourtCase = async (dataSource: DataSource, courtCaseId: number): Promise<UpdateResult | Error> => {
+const unlockCourtCase = async (
+  dataSource: DataSource,
+  courtCaseId: number,
+  user: User
```
Comment (jaskaransarkaria) [inline src/services/unlockCourtCase.ts:9]: Same as above

File: src/services/unlockCourtCase.ts [hunk 5]
```diff
@@ -1,19 +1,29 @@
...
+    if (user.canLockExceptions) {
+      setFields.errorLockedByUsername = null
+    }
+    if (user.canLockTriggers) {
+      setFields.triggerLockedByUsername = null
+    }
+
+    query.set(setFields).where("error_id = :id", { id: courtCaseId })
+
+    return await query.execute()
```
Comment (jaskaransarkaria) [inline src/services/unlockCourtCase.ts:26]: I don't think we need the await when you're returning it

File: src/services/resubmitCourtCase/index.ts [hunk 6]
```diff
@@ -1 +1,2 @@
+// eslint-disable-next-line import/prefer-default-export
```
Comment (jaskaransarkaria) [inline src/services/resubmitCourtCase/index.ts:1]: I thought I set this to warning too 🤔

**Review by jaskaransarkaria — APPROVED** [review]

File: src/features/CourtCaseDetails/CourtCaseDetails.stories.tsx (hunk 1 above)
Comment (emadkaramad) [reply src/features/CourtCaseDetails/CourtCaseDetails.stories.tsx]: Good spot. 👍

File: src/services/unlockCourtCase.ts (hunk 5 above)
Comment (emadkaramad) [reply src/services/unlockCourtCase.ts:26]: Good shout! I also remove the try/catch and use `catch()` function instead.

File: src/services/resubmitCourtCase/index.ts (hunk 6 above)
Comment (emadkaramad) [reply src/services/resubmitCourtCase/index.ts:1]: Not sure tbh. My concern with warnings is that it is easy to miss them. We might end up in an inconsistent situation where some modules are exported as default and some are not.

**Pull request merged**